### DISCUSSION TITLE
How does Image Editor from Project Settings work?

### DISCUSSION THREAD

**Discussion by erondpowell** [directus/directus#15037] [Archive]
How does `Configuration > Project Settings > Image Editor` work?
After messing around for ~20 min I can't identify the purpose of this feature.
I'm not sure what "value" is supposed to be or if it is supposed to save something, etc.

https://user-images.githubusercontent.com/62306630/184262237-bc284d90-68db-484b-bc04-91f431958c55.mov

**Comment by azrikahar** [comment]
It was recently added in #14786, which allows users to add custom aspect ratios 👍

For example if a user would like to have a custom aspect ratio of `16:10`, they can add:

- Text: `16:10`
- Value: `1.6` (this is derived from 16/10, which equals `1.6`. If the intended aspect ratio is `16:9`, the value would be 16 divided by 9, rounded to `1.7778`.)

Understandably the `Value` isn't very obvious here, and it was brought up and discussed with the original PR submitter here: https://github.com/directus/directus/pull/14786#issuecomment-1203930347, to potentially enhance it  as 2 number input instead so that it'll map to the ratios, rather than doing this calculation manually. 

EDIT: another potential enhancement would probably to validate the user input here since I'm not entirely sure what `200-200` would behave here 🤔 or perhaps the input should've prevented non number values.

**Reply by erondpowell** [reply]
@azrikahar
After trying to input `16:9`, I noticed the value box will not let me type in a `:`.. so it only takes fractions??
Oddly, I can also type in a `+` symbol. What's that symbol used for? Perhaps its just a bug?

https://user-images.githubusercontent.com/62306630/184701549-7fce35de-f92c-4038-9a7d-76f04e60f69c.mov

**Reply by azrikahar** [reply]
> After trying to input `16:9`, I noticed the value box will not let me type in a `:`.. so it only takes fractions??

Yeah currently it's only meant for floats/fractions 👍

> Oddly, I can also type in a `+` symbol. What's that symbol used for? Perhaps its just a bug?

It seems to actually be the default behavior for inputs of `type="number"` (which means it should be the default browser behavior here, rather than something Directus is modifying), at least according to the specs: https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#valid-floating-point-number. I believe it's meant for exponents such as `2e+5` and `2e-5`, but the plus sign is probably usually omitted as it can be written as `2e5` instead.

**Reply by erondpowell** [reply]
> I believe it's meant for exponents such as `2e+5` and `2e-5`, but the plus sign is probably usually omitted as it can be written as `2e5` instead.

Hmm, is this a behavior we *would* want to try and limit? 
Seems strange imo to be able to input a massive exponent in as an aspect ratio.

**Reply by azrikahar** [reply]
Agreed that it's strange, but that may require an option in v-input to restrict it for float input 🤔

Perhaps the planned change to make it into a two inputs to enter _integers_ (let's say for `16:9` aspect ratio, the user should enter `16` on first integer input, `9` on second integer input) will still be for the best here, so we won't have to bloat the float input just to fit in this use case 👍
